ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/WrappingAuditAuthenticator.java

```diff
@@ -26,35 +27,36 @@
 import org.apache.cassandra.auth.AuthenticatedUser;
 import org.apache.cassandra.auth.IAuthenticator;
 import org.apache.cassandra.auth.IResource;
+import org.apache.cassandra.auth.IRoleManager;
 import org.apache.cassandra.exceptions.AuthenticationException;
 import org.apache.cassandra.exceptions.ConfigurationException;
 import org.apache.cassandra.utils.FBUtilities;
 
-public class WrappingAuditAuthenticator implements IAuthenticator
+public class WrappingAuditAuthenticator implements IAuditAuthenticator
```

Comment:
Hmm, it feels wrong to implemenet the IAuditAuthenticator interface here. I guess you did this because you want access to the xxxOptions() methods?

...but isn't that the wrong reason? Wouldn't it be more correct to check for instanceof WrappingAuditAuthenticator in the AuditRoleManager? But, on the other hand, then we'd have to check for instanceof AuditPasswordAuthenticator as well...? Let's talk...

Reply:
We'll have a chat about this I guess.